src/http.cc

```diff
@@ -1955,6 +1955,7 @@ HttpStateData::httpBuildRequestHeader(HttpRequest * request,
         snprintf(bbuf, BBUF_SZ, "%s=\"Surrogate/1.0\"", Config.Accel.surrogate_id);
 #endif
         strListAdd(&strSurrogate, bbuf, ',');
+        hdr_out->delById(Http::HdrType::SURROGATE_CAPABILITY);
         hdr_out->putStr(Http::HdrType::SURROGATE_CAPABILITY, strSurrogate.termedBuf());
```

Comment:
As we discussed, this addition could theoretically overflow the resulting `MemBuf` where the `HttpHeader` is packed by `HttpStateData::buildRequestPrefix()`. However, this PR should not cause this because it simply removes the `SURROGATE_CAPABILITY` copy, thus decreasing the header size. For the record, the `MemBuf` in scope is created by `HttpStateData::sendRequest()` and has `max_capacity=MEM_BUF_MAX_SIZE`, i.e., 2GB - that is more than enough for storing any HttpHeader.
It looks like `SURROGATE_CAPABILITY` may be infinitely accumulated due to loops, i.e., the code probably needs some protections as those below for `X_FORWARDED_FOR`. But, IIRC, we decided to ignore such problems in this PR.

Reply:
I agree that solving any overflows is out of scope here.

> the code probably needs some protections as those below for X_FORWARDED_FOR.

Those XFF protections are for String/strListAdd() overflow rather than MemBuf/pack() overflow. Let's ignore 2GB MemBuf overflows for now. Can this code overflow String (e.g., if there is a forwarding loop with XFF and Via disabled)?

Reply:
I think `strListAdd()` can overflow `strSurrogate` if the received SurrogateCapability size is, e.g., `String::SizeMax_ - 1`. This looks unlikely, since `String::SizeMax_ ` is 3*64k.

Reply:
Unknown bugs notwithstanding, can we assert that strSurrogate overflow is _impossible_ (instead of the proposed "unlikely")?

Reply:
I think it is impossible with default request_header_max_size - 64K.